pkg/verifier/cosign/trustpolicy.go

```diff
@@ -0,0 +1,273 @@
+	scopes       []string
+	localKeys    map[PKKey]keymanagementprovider.PublicKey
+	config       TrustPolicyConfig
+	verifierName string
+	isKeyless    bool
+}
+
+type TrustPolicy interface {
+	GetName() string
+	GetKeys(string) (map[PKKey]keymanagementprovider.PublicKey, error)
+	GetScopes() []string
+	GetCosignOpts(context.Context) (cosign.CheckOpts, error)
+}
+
+const (
+	fileProviderName   string = "file"
+	DefaultRekorURL    string = "https://rekor.sigstore.dev"
+	DefaultTLogVerify  bool   = true
+	DefaultCTLogVerify bool   = true
+)
+
+// CreateTrustPolicy creates a trust policy from the given configuration
+// returns an error if the configuration is invalid
+// reads the public keys from the file path
+func CreateTrustPolicy(config TrustPolicyConfig, verifierName string) (TrustPolicy, error) {
+	if err := validate(config, verifierName); err != nil {
+		return nil, err
+	}
+
+	keyMap := make(map[PKKey]keymanagementprovider.PublicKey)
+	for _, keyConfig := range config.Keys {
+		// check if the key is defined by file path or by key management provider
+		if keyConfig.File != "" {
+			pubKey, err := loadKeyFromPath(keyConfig.File)
+			if err != nil {
+				return nil, re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail(fmt.Sprintf("trust policy %s failed: failed to load key from file %s", config.Name, keyConfig.File)).WithError(err)
+			}
+			keyMap[PKKey{Provider: fileProviderName, Name: keyConfig.File}] = keymanagementprovider.PublicKey{Key: pubKey, ProviderType: fileProviderName}
+		}
+	}
+
+	if config.Keyless.RekorURL == "" {
+		config.Keyless.RekorURL = DefaultRekorURL
+	}
+
+	if config.TLogVerify == nil {
+		config.TLogVerify = utils.MakePtr(DefaultTLogVerify)
+	}
+
+	if config.Keyless != (KeylessConfig{}) && config.Keyless.CTLogVerify == nil {
+		config.Keyless.CTLogVerify = utils.MakePtr(DefaultCTLogVerify)
+	}
+
+	return &trustPolicy{
+		scopes:       config.Scopes,
+		localKeys:    keyMap,
+		config:       config,
+		verifierName: verifierName,
+		isKeyless:    config.Keyless != KeylessConfig{},
+	}, nil
+}
+
+// GetName returns the name of the trust policy
+func (tp *trustPolicy) GetName() string {
+	return tp.config.Name
+}
+
+// GetKeys returns the public keys defined in the trust policy
+func (tp *trustPolicy) GetKeys(namespace string) (map[PKKey]keymanagementprovider.PublicKey, error) {
+	keyMap := make(map[PKKey]keymanagementprovider.PublicKey)
+	// preload the local keys into the map of keys to be returned
+	for key, pubKey := range tp.localKeys {
+		keyMap[key] = pubKey
+	}
+
+	for _, keyConfig := range tp.config.Keys {
+		// if the key is defined by file path, it has already been loaded into the key map
+		if keyConfig.File != "" {
+			continue
+		}
+		// must prepend namespace to key management provider name if not provided since namespace is prepended during key management provider intialization
+		namespacedKMP := prependNamespaceToKMPName(keyConfig.Provider, namespace)
```

Comment:
my concern is in multi-tenancy scenario, how to distinguish between cluster-wide and empty namespace.

Reply:
@binbin-li can we discuss this offline? I'd like to make sure this aligns with the multi-tenancy expectation.

Reply:
synced offline, we'll keep the prepend logic in this PR, will address both cosign and notation verifiers in multi-tenancy pr.